climada_petals/hazard/tc_tracks_forecast.py

```diff
@@ -529,3 +539,111 @@ def _check_variable(var, n_ens, varname=None):
+        df_groupby = df.groupby(
+            ["disturbance_no", "baseTime", "basin", "cycloneNumber", "member"],
+            sort=False,
+            dropna=False,
+        )
+        instance = cls()
+        instance.data = [cls._fcastdf_to_ds(subdf) for _, subdf in df_groupby]
+
+        return instance
+
+    @staticmethod
+    def _cxml_to_df(
+        cxml_path: str, xsl_path: str = None, basin_env_pressures: dict = None
+    ):
+        """Read a cxml v1.1 file; may not work on newer specs."""
+        try:
+            import lxml.etree as et
+            import io
+        except ModuleNotFoundError as missing_module:
+            LOGGER.exception("%s, Please install it manually", str(missing_module))
+            raise
+
+        if xsl_path is None:
+            xsl_path = str(CXML2CSV_XSL)
+
+        xsl = et.parse(xsl_path)
+        xml = et.parse(cxml_path)
+        transformer = et.XSLT(xsl)
+        csv_string = str(transformer(xml))
+
+        all_storms_df = pd.read_csv(
+            io.StringIO(csv_string),
+            dtype={
+                "member": "Int64",
+                "cycloneNumber": "Int64",
+                "hour": "Int64",
+                "cycloneName": "object",
+                "id": "object",
+            },
+            parse_dates=["baseTime", "validTime"],
+            infer_datetime_format=True,
+        )
+
+        all_storms_df.dropna(
+            subset=["validTime", "latitude", "longitude"], how="any", inplace=True
+        )
+
+        if basin_env_pressures is None:
+            basin_env_pressures = BASIN_ENV_PRESSURE_CXML
+
+        default_env_pressure = all_storms_df.basin.replace(basin_env_pressures)
+
+        all_storms_df["is_named_storm"] = -all_storms_df["cycloneName"].isna()
+        default_name = (
+            all_storms_df["cycloneNumber"].astype(str) + " - " + all_storms_df["basin"]
+        )
+
+        all_storms_df.fillna(
+            {"cycloneName": default_name, "lastClosedIsobar": default_env_pressure},
+            inplace=True,
+        )
+
+        return all_storms_df
+
+    @staticmethod
+    def _fcastdf_to_ds(track_as_df: pd.DataFrame):
+        """Convert a given subdataframe into an xr.Dataset"""
+
+        if pd.isna(track_as_df["member"].iloc[0]):
+            sid = track_as_df["id"].iloc[0]
+        else:
+            sid = '{}_{}'.format(track_as_df["id"].iloc[0], track_as_df["member"].iloc[0])
+
+        return xr.Dataset(
+            data_vars={
+                "max_sustained_wind": ("time", track_as_df["maximumWind"].values),
+                "central_pressure": ("time", track_as_df["minimumPressure"].values),
+                "hour": ("time", track_as_df["hour"].values),
+                "radius_max_wind": ("time", track_as_df["maximumWindRadius"].values),
+                "environmental_pressure": (
+                    "time",
+                    track_as_df["minimumPressure"].values,
```

Comment:
@mmyrte here we are actually setting environmental pressure equal to central pressure, which I think then leads the TC wind field to be zero. Shouldn't we assign env pressure = `track_as_df['lastClosedIsobar']` here (lastClosedIsobar is where env pressure info are saved above) ?. Also why the name _lastClosedIsobar_ ? I could not find anything names like this in any of the cxml I looked into.

Reply:
Ah, good catch, thanks. I actually only used the MSW data, so I didn't notice it during downstream processing.

The `lastClosedIsobar` is from [this sample CXML file](http://www.bom.gov.au/cyclone/cxmlinfo/CXML_the_works.xml) from <http://www.bom.gov.au/cyclone/cxmlinfo/>. Normally, the env pressure is not recorded in ECMWF forecasts 😢